tests/Helpers/LogAnalyticsHelper.cs

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Net;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+using Azure.Monitor.Ingestion;
+using AzureMcp.Services.Azure.Authentication;
+using AzureMcp.Services.Interfaces;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace AzureMcp.Tests.Helpers;
+
+/// <summary>
+/// Helper class for sending logs to Azure Log Analytics using the Azure Monitor Ingestion SDK.
+/// </summary>
+/// <remarks>
+/// Initializes a new instance of the LogAnalyticsHelper class.
+/// </remarks>
+public class LogAnalyticsHelper(
+    string workspaceName,
+    string subscription,
+    IMonitorService monitorService,
+    string? tenantId = null,
+    string logType = "TestLogs_CL",
+    ILogger? logger = null)
+{
+    private readonly string _workspaceName = workspaceName;
+    private readonly string _subscription = subscription;
+    private readonly string _logType = logType;
+    private readonly string? _tenantId = tenantId;
+    private readonly TokenCredential _credential = new CustomChainedCredential(tenantId);
+    private readonly IMonitorService _monitorService = monitorService ?? throw new ArgumentNullException(nameof(monitorService));
+    private readonly ILogger _logger = logger ?? NullLogger.Instance;
+    private readonly SemaphoreSlim _clientInitLock = new(1, 1);
+    private string? _workspaceId;
+    private LogsIngestionClient? _logsIngestionClient;
+
+    private async Task<string> GetWorkspaceIdAsync()
+    {
+        if (!string.IsNullOrEmpty(_workspaceId))
+        {
+            return _workspaceId;
+        }
+
+        // Get workspace info using the monitor service
+        var workspaces = await _monitorService.ListWorkspaces(_subscription, _tenantId);
+        var workspace = workspaces.FirstOrDefault(w => w.Name.Equals(_workspaceName, StringComparison.OrdinalIgnoreCase))
```

Comment:
Do we expect to have just 1 workspace returned from the `ListWorkspaces` call?

Reply:
Yes